### PR TITLE
added logic for showing warning when reserved keywords ae sent in pro…

### PR DESCRIPTION
**Pull request by ruchiramoitra**
…perties/traits

## Description of the change

> Description here

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

---
This change is [Reviewable](https://reviewable.io/reviews/rudderlabs/rudder-sdk-js/181)

### CODE REVIEW

File: utils/constants.js [hunk 1]
```diff
@@ -79,6 +90,7 @@ const MAX_WAIT_FOR_INTEGRATION_LOAD = 10000;
 const INTEGRATION_LOAD_CHECK_INTERVAL = 1000;
 
 export {
+  reservedPropKeywords,
```
Comment (arnab-p) [inline utils/constants.js:93]: We should maintain the naming convention - `ReservedPropertyKeywords`

File: analytics.js [hunk 2]
```diff
@@ -497,6 +498,11 @@ class Analytics {
     if (!properties) {
       properties = {};
     }
+    Object.keys(properties).forEach((property) => {
+      if (reservedPropKeywords.indexOf(property) >= 0) {
+        console.error(`Warning! : Reserved keyword used --> ${property} with page call for name --> ${name}`);
```
Comment (arnab-p) [inline analytics.js:503]: Should we use `logger` here and use a `warning` log level instead of `console.error`?

**Review by arnab-p — CHANGES_REQUESTED** [review]

File: analytics.js [hunk 3]
```diff
@@ -529,7 +535,11 @@ class Analytics {
     } else {
       rudderElement.setProperty({});
     }
-
+    Object.keys(properties).forEach((property) => {
```
Comment (sayan-mitra) [inline analytics.js:538]: add a util function and pass props/traits as parameter ?
also can this check be done at a central loc like in https://github.com/rudderlabs/rudder-sdk-js/blob/production/analytics.js#L638

File: analytics.js [hunk 4]
```diff
@@ -529,7 +535,11 @@ class Analytics {
...
+      if (reservedPropKeywords.indexOf(property) >= 0) {
+        console.error(`Warning! : Reserved keyword used --> ${property} with track call for event --> ${event}`);
```
Comment (sayan-mitra) [inline analytics.js:540]: logger.error should be fine I think
other levels probably needs a wrapper/polyfill for support in older browsers

**Review by sayan-mitra — CHANGES_REQUESTED** [review]
Give the PR on staging

**Comment by sayan-mitra** [comment]
closing this as #184  has the changes

**Pull request closed without merging**